src/start_server.ts

```diff
@@ -181,6 +161,179 @@ function startWithPort(userOptions: ServerOptions) {
       });
     }
   }
+}
+
+/**
+ * Determines whether a protocol requires HTTPS
+ * @param {string} protocol Protocol to evalulate.
+ * @returns {boolean}
+ */
+function isHttps(protocol: string) {
+  return ['https/1.1', 'https', 'http2', 'h2'].indexOf(protocol) > -1;
+}
+
+/**
+ * Gets the URLs for the main and component pages
+ * @param {ServerOptions} options
+ * @returns {{serverUrl: {protocol: (string|string), hostname: string, port: string}, componentUrl: url.Url}}
+ */
+function getServerUrls(options: ServerOptions) {
+  let serverUrl = {
+    protocol: isHttps(options.protocol) ? 'https' : 'http',
+    hostname: options.hostname,
+    port: `${options.port}`,
+  };
+  let componentUrl: url.Url = Object.assign({}, serverUrl);
+  componentUrl.pathname = `components/${options.packageName}/`;
+  return {serverUrl, componentUrl};
+}
+
+/**
+ * Handles server-ready tasks (prints URLs and opens browser)
+ * @param {ServerOptions} options
+ */
+function handleServerReady(options: ServerOptions) {
+  let {serverUrl, componentUrl} = getServerUrls(options);
```

Comment:
Ok. Any particular reason why tsconfig is targeting ES6? If we switch to ES5 (which Node 4 supports), then we should be able to freely use any ES6 feature. It's trivial to rework this, but just wanted to make sure we're not unnecessarily catering to older versions of Node.